docs/source/daml/reference/choices.rst

```diff
@@ -91,9 +136,11 @@ Non-consuming choices
 
 - ``nonconsuming`` keyword. Optional.
 - Makes a choice non-consuming: that is, exercising the choice does not archive the contract.
+- Contracts created in the choice body are known only to the signatories and controllers of the contracts and not made known to the observers of the contract on which the choice was exercised.
+
+  By default, choices are *preconsuming*: when a choice on a contract is exercised, that contract instance is *archived*. Archived means that it's permanently marked as being inactive, and no more choices can be exercised on it, though it still exists on the ledger.
 
-  By default, choices are *consuming*: when a choice on a contract is exercised, that contract instance is *archived*. Archived means that it's permanently marked as being inactive, and no more choices can be exercised on it, though it still exists on the ledger.
-- This is useful in the many situations when you want to be able to exercise a choice more than once.
+- Both *preconsuming* and *postconsuming* behaviors are useful in the many situations when you want to be able to exercise a choice more than once.
```

Comment:
I'm not sure I understand this correctly but what I understand sounds wrong. `preconsuming` and `postconsuming` still archive the contract and hence can only ve exercised once. `nonconsuming` has the property you're describing.

Reply:
Yeah, I guess I messed up while re-ordering some paragraph. Thanks.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4134/commits/a133907a47c005289fce27fb37e67c2798774ea2